pkg/proxy/iptables/proxier.go

```diff
@@ -788,11 +791,25 @@ func (proxier *Proxier) execConntrackTool(parameters ...string) error {
 // assumes proxier.mu is held
 func (proxier *Proxier) syncProxyRules() {
 	if proxier.throttle != nil {
-		proxier.throttle.Accept()
+		// NOTE: using non-blocking TryAccept.
+		// All other updates should update the datastructure
+		// and do the appropriate thing to batch when we are not syncing.
+		// Lastly, the sync timer will be set to ensure we catch the last update.
+		if proxier.throttle.TryAccept() == false {
+			duration := time.Since(proxier.lastSync)
+			if duration < proxier.minSyncPeriod {
```

Comment:
You have to wait the min-sync-period not full-sync-period.  

> can we just drop the no-op updates before applying any rate limits?

Potentially yes, there is a separate patch for that #36006